src/OutputFile.zig

```diff
@@ -231,86 +233,53 @@ pub fn init(options: Options) OutputFile {
                         try root_dir.makePath(parent);
                     }
                 }
             }
 
-            var handled_file_not_found = false;
-            while (true) {
-                var path_buf: bun.PathBuffer = undefined;
-                JSC.Node.NodeFS.writeFileWithPathBuffer(&path_buf, .{
-                    .data = .{ .buffer = .{
-                        .buffer = .{
-                            .ptr = @constCast(value.bytes.ptr),
-                            .len = value.bytes.len,
-                            .byte_len = value.bytes.len,
-                        },
-                    } },
-                    .encoding = .buffer,
-                    .mode = if (f.is_executable) 0o755 else 0o644,
-                    .dirfd = bun.toFD(root_dir.fd),
-                    .file = .{ .path = .{
-                        .string = JSC.PathString.init(rel_path),
-                    } },
-                }).unwrap() catch |err| switch (err) {
-                    error.FileNotFound, error.ENOENT => {
-                        if (handled_file_not_found) return err;
-                        handled_file_not_found = true;
-                        try root_dir.makePath(
-                            std.fs.path.dirname(rel_path) orelse
-                                return err,
-                        );
-                        continue;
+            var path_buf: bun.PathBuffer = undefined;
+            _ = try JSC.Node.NodeFS.writeFileWithPathBuffer(&path_buf, .{
+                .data = .{ .buffer = .{
+                    .buffer = .{
+                        .ptr = @constCast(value.bytes.ptr),
+                        .len = value.bytes.len,
+                        .byte_len = value.bytes.len,
                     },
-                    else => return err,
-                };
-                break;
-            }
+                } },
+                .encoding = .buffer,
+                .mode = if (f.is_executable) 0o755 else 0o644,
+                .dirfd = bun.toFD(root_dir.fd),
+                .file = .{ .path = .{
+                    .string = JSC.PathString.init(rel_path),
+                } },
+            }).unwrap();
 
             return rel_path;
         },
         .move => |value| {
-            _ = value;
-            // var filepath_buf: bun.PathBuffer = undefined;
-            // filepath_buf[0] = '.';
-            // filepath_buf[1] = '/';
-            // const primary = f.dest_path[root_dir_path.len..];
-            // bun.copy(u8, filepath_buf[2..], primary);
-            // var rel_path: []const u8 = filepath_buf[0 .. primary.len + 2];
-            // rel_path = value.pathname;
-
-            // try f.moveTo(root_path, @constCast(rel_path), bun.toFD(root_dir.fd));
-            {
-                @panic("TODO: Regressed behavior");
-            }
-
-            // return primary;
+            try f.moveTo(root_dir_path, value.pathname, bun.toFD(root_dir.fd));
```

Comment:
kinda scary this branch wasn't compiled before. is `moveTo` tested?

Reply:
oh wait, I see it was compiled.

Reply:
going to be honest. i still don't know how to actually hit this code path. it is likely something we will completely delete one day.